shoop/admin/modules/orders/json_order_creator.py

```diff
@@ -121,27 +136,93 @@ def _process_lines(self, source, state):
             except Exception as exc:  # pragma: no cover
                 self.add_error(exc)
 
-    def _initialize_source_from_state(self, state, creator):
-        customer_data = state.pop("customer", None) or {}
-        shop_data = state.pop("shop", None) or {}
-        methods_data = state.pop("methods", None) or {}
-        customer = self.safe_get_first(Contact, pk=customer_data.get("id"))
-        shop = self.safe_get_first(Shop, pk=shop_data.pop("id", None))
+    def _create_contact_from_address(self, billing_address, is_company):
+        name = billing_address.get("name", None)
+        if not name:
+            self.add_error(
+                ValidationError(_("Name is required for customer"), code="no_name")
+            )
 
-        if not customer:
-            self.add_error(ValidationError(_("Please choose a valid customer."), code="no_customer"))
+        phone = billing_address.get("phone", "")
+        email = billing_address.get("email", "")
+        fields = {"name": name, "phone": phone, "email": email}
+        if is_company:
+            tax_number = billing_address.pop("tax_number", None)
+            if not tax_number:
+                self.add_error(
+                    ValidationError(_("Tax number is not set for company."), code="no_tax_number")
+                )
+                return
+
+            fields.update({"tax_number": tax_number})
+            customer = CompanyContact.objects.create(**fields)
+        else:
+            customer = PersonContact.objects.create(**fields)
+        return customer
 
+    def _create_address(self, address):
+        address_fields = [
+            "name", "tax_number", "phone", "email", "street", "street2", "postal_code", "city", "region", "country"
+        ]
+        fields = {}
+        for field in address_fields:
+            fields[field] = address.get(field, "")
+
+        address = Address.objects.filter(**fields).first()
```

Comment:
Order should cache addresses. The current implementation causes database to be flooded with addresses once used with one order. I think address should be "customers property" where only direct relation lives. All other places (if there is a fear for address data to change) should only cache these values for themselves. Also once order is made, it's data should be set in stone. If someone modifies something related to order, it should not change the order contents. ever. For me, this current design is faulty.

tl;dr; remove `Order.billing_address` and `Order.shipping_address` and replace them with address properties (`shipping_street`, `shipping_zip` etc)